### PR TITLE
ENH: add tilt angle option for all spiral scans

### PR DESCRIPTION
**Pull request by klauer**
cc @xiaojinghuang

### CODE REVIEW

File: bluesky/plan_patterns.py [hunk 1]
```diff
@@ -29,6 +30,8 @@ def spiral(x_motor, y_motor, x_start, y_start, x_range, y_range, dr, nth):
         Delta radius
     nth : float
         Number of theta steps
+    tilt : float, optional
+        Tilt angle in degrees, default 0.0
```
Comment (danielballan) [inline bluesky/plan_patterns.py]: Standard practice in scipy is to make the interfaces radians, not degrees. Is that acceptable?

Comment (klauer) [inline bluesky/plan_patterns.py]: Switched to radians.

**Comment by danielballan** [comment]
:+1:

File: bluesky/plan_patterns.py [hunk 2]
```diff
@@ -105,7 +109,8 @@ def spiral_fermat(x_motor, y_motor, x_start, y_start, x_range, y_range, dr,
         x = radius * np.cos(angle)
         y = radius * np.sin(angle)
 
-        if abs(x) <= half_x and abs(y) <= half_y:
+        if (abs(x - y / tilt_tan) <= half_x and
```
Comment (tacaswell) [inline bluesky/plan_patterns.py]: Can you add `()` here just to be sure of what the order of operations is?

Comment (klauer) [inline bluesky/plan_patterns.py]: Added

**Pull request merged**